plonky3/src/params.rs

```diff
@@ -1,98 +1,19 @@
-//! The concrete parameters used in the prover
-//! Inspired from [this example](https://github.com/Plonky3/Plonky3/blob/6a1b0710fdf85136d0fdd645b92933615867740a/keccak-air/examples/prove_goldilocks_keccak.rs#L57)
+use p3_commit::{Pcs, PolynomialSpace};
+use p3_uni_stark::StarkGenericConfig;
+use powdr_number::FieldElement;
 
-use lazy_static::lazy_static;
+pub type Plonky3Field<T> =
+    <<MyPcs<T> as Pcs<Challenge<T>, Challenger<T>>>::Domain as PolynomialSpace>::Val;
+pub type MyPcs<T> = <<T as FieldElementMap>::Config as StarkGenericConfig>::Pcs;
+pub type Challenge<T> = <<T as FieldElementMap>::Config as StarkGenericConfig>::Challenge;
+pub type Challenger<T> = <<T as FieldElementMap>::Config as StarkGenericConfig>::Challenger;
 
-use p3_challenger::DuplexChallenger;
-use p3_commit::ExtensionMmcs;
-use p3_dft::Radix2DitParallel;
-use p3_field::{extension::BinomialExtensionField, Field};
-use p3_fri::{FriConfig, TwoAdicFriPcs};
-use p3_goldilocks::MdsMatrixGoldilocks;
-use p3_merkle_tree::FieldMerkleTreeMmcs;
-use p3_poseidon::Poseidon;
-use p3_symmetric::{PaddingFreeSponge, TruncatedPermutation};
-use p3_uni_stark::StarkConfig;
+pub trait FieldElementMap: FieldElement {
+    type Config: StarkGenericConfig + Sync;
 
-use rand::{distributions::Standard, Rng, SeedableRng};
+    fn to_p3_field(&self) -> Plonky3Field<Self>;
```

Comment:
```suggestion
    fn into_p3_field(self) -> Plonky3Field<Self>;
```

It makes sense to move `self` in this case